SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -238,18 +243,32 @@ void USpatialReceiver::OnRemoveEntity(const Worker_RemoveEntityOp& Op)
 		return;
 	}
 
-	RemoveActor(Op.entity_id);
+	if (LoadBalanceEnforcer != nullptr)
+	{
+		LoadBalanceEnforcer->OnEntityRemoved(Op);
+	}
+
 	OnEntityRemovedDelegate.Broadcast(Op.entity_id);
 }
 
 void USpatialReceiver::OnRemoveComponent(const Worker_RemoveComponentOp& Op)
 {
+	if (Op.component_id == SpatialConstants::UNREAL_METADATA_COMPONENT_ID)
+	{
+		RemoveActor(Op.entity_id);
```

Comment:
What would happen if this and a normal unreal component is removed in the same frame?
If this is processed first, the actor would be destroyed, then all hell breaks loose

Reply:
it should be fine because all other remove components will be queued as before and handled correctly if the actor is gone in the remove component ops function

Reply:
hmm actually that is only for remove entity

Reply:
although the comment below is hinting that processremovecomponent would handle this case correctly by dropping them

Reply:
Oh, that comment is outdated then, as we drop them when DropQueuedRemoveComponents is called.

Reply:
misleading comments are my favourite

Reply:
it will just not do anything in process remove if there is no more actor as there is an if actor check, so I think this is fine